Make sysctl calls on FreeBSD

sambler opened the pull request:
The OS module get_unique_id and get_processor_name rely on linux files which don't exist on a standard FreeBSD install, make sysctl calls to get the required data.

I expect OpenBSD and NetBSD would require similar changes.

AThousandShips commented:
You didn't update your branch, it is still about 18 months out of date, please update your branch with `git rebase -i master` after updating your `master` branch to the latest version

AThousandShips commented on platform/linuxbsd/os_linuxbsd.cpp:
```diff
@@ -139,17 +143,36 @@ void OS_LinuxBSD::initialize_joypads() {
+	char buf[4096];
+	memset(buf,0,sizeof(buf));
+	size_t len = sizeof(buf)-1;
+	if ( sysctl(mib, 2, buf, &len, 0x0, 0) == -1)
+		return String();
```
```suggestion
	if (sysctl(mib, 2, buf, &len, 0x0, 0) == -1) {
		return String();
	}
```

AThousandShips commented:
You rebased incorrectly, please reset with `git reset --hard 6a9caf6` and try again

akien-mga commented:
There are still style issues that would be fixed by clang-format (or checking surrounding code and using the same style).

And the commits are still not linked to your GH account.

sambler commented:
@akien-mga hi, when I look at the commit history it says I authored and committed the changes, so I'm not sure how its not linked to my account, or what I would need to change.

AThousandShips commented:
See: [Why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) for more info.

sambler commented on platform/linuxbsd/os_linuxbsd.cpp:
```diff
@@ -145,17 +149,37 @@ void OS_LinuxBSD::initialize_joypads() {
+		char buf[4096];
+		memset(buf, 0, sizeof(buf));
+		size_t len = sizeof(buf) - 1;
+		if (sysctl(mib, 2, buf, &len, 0x0, 0) != -1) {
```
The man page only describes the values as a string, there appears to be no specification of ascii or utf requirements.

sambler commented:
Seems there was a capital letter in my github email that makes it different to the commit info.

akien-mga reviewed (approved):
Didn't test, but assuming it works, the changes seem fine.

Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

While doing so, you would also get a change to update your Git email to make it match your GitHub account's (with `git commit --amend --reset-author` after changing your local or global Git email).

akien-mga commented:
You added unexpected changes, you probably rebased on your old local `master` branch again instead of upstream.

sambler commented:
I think it looks right this time

akien-mga commented:
Yes, looks great!

Faless commented on platform/linuxbsd/os_linuxbsd.cpp:
```diff
@@ -165,6 +189,7 @@ String OS_LinuxBSD::get_processor_name() const {
 			return line.split(":")[1].strip_edges();
 		}
 	}
+#endif
```
Shouldn't the `#endif` go below?

```
ERR_FAIL_V_MSG("", String("Couldn't get the CPU model name from `/proc/cpuinfo`. Returning an empty string."));
```

Otherwise that's unreachable code for FreeBSD (since we always return before it)

sambler replied:
Actually, I think I should negate the `if` so that it returns a valid value and lets it fall through to this message when there is no result to return.

Faless replied:
Ah, I overlooked that if :eyes:, sorry.

akien-mga reviewed (approved):
Looks good! Could you squash the commits?

sambler committed:
Make sysctl calls on FreeBSD

The OS module get_unique_id and get_processor_name rely
on linux files which don't exist on a standard FreeBSD install,
make sysctl calls to get the required data.

sambler commented:
@akien-mga should be good to go now. Note that I also added the `String::utf8` change which I didn't push earlier.

akien-mga commented:
Cherry-picked for 4.2.2.
